### PR TITLE
cdc,pkg(ticdc): refactor pdtime.Clock

### LINKED ISSUE #4216: TiKVStorage and TimeAcquirer can be shared
As title.

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #4216

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/pdtime/clock.go [hunk 1]
```diff
@@ -72,47 +73,48 @@ func (c *TimeAcquirerImpl) Run(ctx context.Context) {
...
+				c.mu.err = err
 				c.mu.Unlock()
 			}
 		}
 	}
 }
 
-// CurrentTimeFromCached return current time from pd cache
-func (c *TimeAcquirerImpl) CurrentTimeFromCached() (time.Time, error) {
+// CurrentTime return current time from pd cache
```
Comment (asddongmen) [inline pkg/pdtime/clock.go:92]:
```suggestion
// CurrentTime returns current time from timeCache
```

File: pkg/pdtime/clock.go [hunk 2]
```diff
@@ -28,41 +28,42 @@ import (
...
-	return &TimeAcquirerImpl{
+// NewClock return a new PDClock
+func NewClock(pdClient pd.Client) *PDClock {
+	return &PDClock{
 		pdClient: pdClient,
+		stopCh:   make(chan struct{}, 1),
 	}
 }
 
 // Run will get time from pd periodically to cache in pdPhysicalTimeCache
```
Comment (asddongmen) [inline pkg/pdtime/clock.go:57]:
```suggestion
// Run will get time from pd periodically to cache in timeCache
```

File: pkg/pdtime/clock.go [hunk 3]
```diff
@@ -28,41 +28,42 @@ import (
...
+// PDClock cache time get from PD periodically and cache it
+type PDClock struct {
+	pdClient pd.Client
+	mu       struct {
+		sync.RWMutex
+		timeCache time.Time
+		err       error
+	}
+	cancel context.CancelFunc
+	stopCh chan struct{}
```
Comment (asddongmen) [inline pkg/pdtime/clock.go:46]: why we need this `stopCh`?

Comment (overvenus) [reply pkg/pdtime/clock.go:46]: Nice catch, I want to synchronize `Stop`.

Comment (asddongmen) [reply pkg/pdtime/clock.go:46]: It seems that `Clock.Stop()` is not thread-safety, will it cause any possible problem?

**Review by hi-rustin — APPROVED** [review]

File: pkg/pdtime/clock.go (hunk 3 above)
Comment (asddongmen) [reply pkg/pdtime/clock.go:46]: > It seems that `Clock.Stop()` is not thread-safety, will it cause any possible problem?

I am wrong about this, Never mind.

**Review by asddongmen — APPROVED** [review]

**Pull request merged**